src/jit/lsraxarch.cpp

```diff
@@ -2323,7 +2324,7 @@ void LinearScan::BuildHWIntrinsic(GenTreeHWIntrinsic* intrinsicTree)
             assert(numArgs != 0);
             assert(numArgs != 1);
 
-            if (info->srcCount >= 2)
+            if (info->srcCount >= 2 && (genActualType(intrinsicTree->TypeGet()) == genActualType(op2->TypeGet())))
```

Comment:
@tannergooding I think we can have a stricter check to avoid handling special cases like `Insert`.

Reply:
Never mind, we cannot have this check.